src/modules/land_detector/LandDetector.h

```diff
@@ -154,26 +153,33 @@ class LandDetector : public ModuleBase<LandDetector>, px4::ScheduledWorkItem
 	systemlib::Hysteresis _ground_contact_hysteresis{true};
 	systemlib::Hysteresis _ground_effect_hysteresis{false};
 
-	struct actuator_armed_s	_arming {};
+	actuator_armed_s _arming {};
 
 private:
-	void Run() override;
 
-	void _check_params(bool force = false);
```

Comment:
Why is `force` removed?

Reply:
Hi Beat, thanks for your review! I removed `force` because `check_params()` is private, so we know nothing outside of the class is utilizing it, and the [only place it now gets called](https://github.com/PX4/Firmware/blob/d96c97ed48813b0b93a01d4b669ee36dc2633104/src/modules/land_detector/LandDetector.cpp#L84) is setting `force` to  `false`, so it has become unneeded.

Reply:
It needs to be called with `force=true` here: https://github.com/PX4/Firmware/pull/12209/files#diff-1fd2cab203a1ba1c362f2dd76a74542fL84. Otherwise `_total_flight_time` does not necessarily get initialized. This only worked because there were other param changes.

Reply:
Hi Beat. So this is broken in current master?  I have to admit that I'm a little confused, in the tests I show above only the `LAND_FLIGHT_T_HI` and `LAND_FLIGHT_T_LO` are changed, no others.  It is also verifiable with no parameter chances in the flight logs below.  I'll gladly revert this change though if it's important.

Reply:
> So this is broken in current master?

Yes. We always have other param changes at the end of a flight, which is why it worked.

Reply:
Hi @bkueng. I pushed up a commit that calls `_check_params(true)` within the start() method so that it gets initialized once as used to be the case prior to 106ee280c512700284b846396f4ea987a4ed9462.

Let me know if you think this is appropriate, and otherwise, if it needs to be forced at each `Run()` call, we can deprecate the bool arg and the `if` statement so that `_check_params()` can be effectively an `_update_params()` call each iteration.

Reply:
@bkueng , moving to the constructor absolutely breaks the detector.  I am removing it and squashing that commit out of existence.

Reply:
Hi @bkueng, placing the call back in the `start()` method works fine and I think will satisfy the concerns.  Latest commit functions fine, flight tested here: https://review.px4.io/plot_app?log=cdc782c2-9999-4f49-a98a-2de786ce5c6f

Let me know if you have any other thoughts! Thanks again for your review!